public/characterCreator.html

```diff
@@ -1 +1,157 @@
+         * @param {number} setstat The level the stat is set to
+         * @return {number} result This is the resulting cost of statpoints for the set stat, negative if its a gain of statpoints
+         */
+        function generateStatCost(setStatLevel) {
+            setStatLevel = parseInt(setStatLevel);
+            if (setStatLevel < 0) { //Checks if setstatlevel is negative if so returns the negative setstatlevel as a statpoint gain
+                return setStatLevel
+            } else {
+                for (loopIterator = 0, result = 0; loopIterator <= setStatLevel; loopIterator++) { // iterates to find the statpoint cost needed to get a stat to the setstat level
+                    result = result + loopIterator;
+                }
+                return result; // returns statpoint cost in the case of it being a non negative.
+            }
+        }
+        /** Returns the net cost of set statlevels with current ammount of statpoitns, returns negative if their is a deficit and positive if there is a suprlus of statpoints
+         * @param {htmlcollection} statCollection This is a collection of elements that contain number values within the class 'stats'
+         * @return {number} workingValue This is a number that is the total statpoint cost being either negative or positive depending on if free statpoints remain.
+         */
+        function calculateDeficit(statCollection) {
+            let workingValue = 0;
+            for (statValue of statCollection) {
+                if (isNaN(statValue.value)) {
+                    return "Invalid entry";
+                }
+                workingValue += generateStatCost(statValue.value);
+            }
+            const setStatPoints = parseInt(document.getElementById('statpoints').value);
+            workingValue = setStatPoints - workingValue;
+            return workingValue;
+        }
+
+        /** Listens to the form submission, takes the values of the form, puts them into json, and puts it into a cookie called "value".
+         * @param {event} event The submission of the 'characterCreator' form, captured and instead of being sent to an endpoint, it is set as the value of the cookie "value".
+         */
+        document.getElementById('characterCreator').onsubmit = function(event) {
+            event.preventDefault();
+            const elements = document.getElementsByClassName("stats");
+            if (calculateDeficit(elements) < 0) {
+                alert("Not enough stat points for selected stats.  Check your remaining points.");
+            } else {
+
+                const data = new FormData(event.target);
+                const value = JSON.stringify(Object.fromEntries(data.entries()));
+
+                //Put this JSON string into a cookie
+                setCookie("value", value, 365);
+
+                //and take us to the character sheet page to see our new character stats
+                window.location.href = "characterSheet.html";
```

Comment:
Have you considered instead of doing it like this, having it be a submit handler on the form and not setting the href? I think that doing the submit handler would be better code style/practice https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event